src/client/datascience/raw-kernel/rawJupyterSession.ts

```diff
@@ -203,15 +203,16 @@ export class RawJupyterSession extends BaseJupyterSession {
     protected shutdownSession(
         session: ISessionWithSocket | undefined,
         statusHandler: Slot<ISessionWithSocket, Kernel.Status> | undefined,
-        isRequestToShutDownRestartSession: boolean | undefined
+        isRequestToShutdownRestartSession: boolean | undefined
     ): Promise<void> {
         // REmove our process exit handler. Kernel is shutting down on purpose
         // so we don't need to listen.
         if (this.processExitHandler) {
             this.processExitHandler.dispose();
             this.processExitHandler = undefined;
         }
-        return super.shutdownSession(session, statusHandler, isRequestToShutDownRestartSession).then(() => {
+        return super.shutdownSession(session, statusHandler, isRequestToShutdownRestartSession).then(() => {
+            sendTelemetryEvent(Telemetry.RawKernelSessionShutdown, undefined, { isRequestToShutdownRestartSession });
```

Comment:
Does this capture the callstack here? I think that's part of what we need to identify why this is occurring.